JavaScript/JavaScript.sublime-syntax

```diff
@@ -103,6 +103,8 @@ variables:
 
   # '@' followed by a pattern like \S but excluding literal '*' and '@'.
   jsdoc_block_tag: \@[^\n\t\f\v *@]+
+  # Tags that are optionally followed by a type expression.
+  jsdoc_block_tag_with_type: \@(?:arg(?:ument|)|param|type(?:def|)|returns?|prop(?:erty|)|enum|implements|member|module|namespace|(?:p|rivate|protected)|throws|exception|yields?|var)\b
```

Comment:
No JSDoc expert here, but is there a mistaken extra `|` in `p|rivate`? And what's the purpose of this uncaptured group around `private|protected`?
Also e.g. `arg(?:ument)?` would be the same as `arg(?:ument|)`, but more readable imo. Or maybe even just write them out as `arg|argument`.

Reply:
> but is there a mistaken extra `|` in `p|rivate`?

This looks wrong. Good catch.

> Also e.g. `arg(?:ument)?` would be the same as `arg(?:ument|)`, but more readable imo.

Yes, that should be the same. I can go with either. No strong feelings.

> Or maybe even just write them out as `arg|argument`.

I could but that's a question I should try to optimize here or not bother with it.

Reply:
> > Or maybe even just write them out as `arg|argument`.
> I could but that's a question of whether I should try to optimize here or not bother with it.

IIRC the engine optimises them internally, whichever of those you write. So pick the one that looks nice for humans.

Reply:
I know there was a discussion about that somewhere but I can't find it now. Are you able to provide a reference for that?

Reply:
I think I've also seen something about this related to the PHP syntax before somewhere here, but I don't know where exactly.

Another thing I noticed from looking at this implementation (I might be wrong because I haven't tested it), is that you push/set into an anynomous context with the first opening `{`, but only pop on `}`. What happens if someone writes
```
/** @type {string  no closing brace here */
```
I guess that might confuse the JSDoc parser, but it is still a valid comment and a valid JS file, right? If it is, you might better add a syntax test for such cases with missing single or double braces. Right now I assume the rest of the file be highlighted as a comment then, until it hits a closing brace somewhere.

Reply:
You are right. This should be handled better. Should handle the comment-end marker at least so that highlighting doesn't continue to the rest of the file.

(Just a note that those type blocks can be multiline so we can't interrupt parsing at the end of the line.)

Reply:
Note: That issue was fixed with later commits.